Implemented Changes to getDataValues

matthewcn56 opened the pull request:
Changed the ttt values of getDataValues to accept xAngle, yAngle, and time

Accidentally changed some of accel gyro sensor as well lol, and am going to finish processData

matthewcn56 committed:
Implemented Changes to getDataValues

Accidentally changed some of accel gyro sensor as well lol, and am going to finish processData

JasonWu2020 commented on Changed_Files/main.c:
```diff
@@ -476,977 +488,1067 @@ void Feature_Extraction_State_0(void* handle, int* ttt_1, int* ttt_2,
+    CDC_Fill_Buffer((uint8_t*)msg, strlen(msg));
+    float xAngle[3] = {0,0,0};
+    float zAngle[3] = {0,0,0};
+    float xRotationRate[3] = {0,0,0};
```
Get rid of this

JasonWu2020 commented on Changed_Files/main.c:
```diff
@@ -476,977 +488,1067 @@ void Feature_Extraction_State_0(void* handle, int* ttt_1, int* ttt_2,
+            }
+        }
+        //Save first motion index to keep track of time
+        int firstMotionIndex = currentMotionIndex;
```
No need

JasonWu2020 commented on Changed_Files/main.c:
```diff
@@ -476,977 +488,1067 @@ void Feature_Extraction_State_0(void* handle, int* ttt_1, int* ttt_2,
+
+        }
+
+        xAngle[count] = fabs(xAngle[count]);
```
No abs

JasonWu2020 commented on Changed_Files/main.c:
```diff
@@ -476,977 +488,1067 @@ void Feature_Extraction_State_0(void* handle, int* ttt_1, int* ttt_2,
+        zAngle[count] = fabs(zAngle[count]);
+
+        //TODO potentially eliminate the last few rotationRates due to shake
+        xRotationRate[count] = values[(currentMotionIndex + lastMotionIndex) / 2].xRotation - values[lastMotionIndex + 2].xRotation;
```
No need

JasonWu2020 commented on Changed_Files/main.c:
```diff
@@ -476,977 +488,1067 @@ void Feature_Extraction_State_0(void* handle, int* ttt_1, int* ttt_2,
+        count++;
+    }
+    for (int i = 0; i < 3; i++) {
+        int a = xRotationRate[i];
```
No need

JasonWu2020 commented on Changed_Files/main.c:
```diff
@@ -476,977 +488,1067 @@ void Feature_Extraction_State_0(void* handle, int* ttt_1, int* ttt_2,
+    //Assign ttt_3 to the time between start and end of motion in milliseconds
+    //TODO Should we still magnify rotation rate values???
+    //TODO DELETE THIS COMMENT @Jason, why do we multiply the two avgs by different constants?
+    int xAngleAvgScaled = (int)(fabs((xAngle[0] + xAngle[1] + xAngle[2]) * 0.66666));
```
No abs, multiply by 0.3333

JasonWu2020 commented on Changed_Files/main.c:
```diff
@@ -476,977 +488,1067 @@ void Feature_Extraction_State_0(void* handle, int* ttt_1, int* ttt_2,
+    //TODO DELETE THIS COMMENT @Jason, why do we multiply the two avgs by different constants?
+    int xAngleAvgScaled = (int)(fabs((xAngle[0] + xAngle[1] + xAngle[2]) * 0.66666));
+    int zAngleAvgScaled = (int)(fabs((zAngle[0] + zAngle[1] + zAngle[2]) * 0.66666));
+    int xRotationRateAvg = (int)(xRotationRate[0] + xRotationRate[1] + xRotationRate[2]) * 0.33333;
```
Delete

JasonWu2020 commented on Changed_Files/main.c:
```diff
@@ -476,977 +488,1067 @@ void Feature_Extraction_State_0(void* handle, int* ttt_1, int* ttt_2,
+    *ttt_1 = xAngleAvgScaled;
+    *ttt_2 = zAngleAvgScaled;
+    //Set ttt_3 equal to time between start and end of motion in milliseconds
+    *ttt_3 = lastMotionIndex-firstMotionIndex * 50;
```
Prob do *10

matthewcn56 committed:
Addressed changes within updateValues

Will also address merge conflict within accel gyro sensor handler by accepting your version instead of mine

matthewcn56 committed:
more fixed changes

JasonWu2020 reviewed (commented):
Yup we good

JasonWu2020 reviewed (commented):
Okay is this working

matthewcn56 committed:
Merge branch 'master' into matthew-getDataValues
